cfme/tests/configure/test_landing_page.py

```diff
@@ -1,159 +1,232 @@
-    # except NoSuchElementException:
-    #     browser.start(url_key="{}/configuration/index".format(appliance.server.address))
-    #     view.visualstartpage.show_at_login.fill(value)
-    #     view.save.click()
-    login_page = navigate_to(appliance.server, 'LoginScreen')
-    login_page.login_admin()
-    logged_in_page = navigate_to(appliance.server, 'LoggedIn')
-    return logged_in_page.is_displayed
+@pytest.fixture(scope="module")
+def set_default_page(my_settings):
+    default_page = my_settings.visual.login_page
+    yield
+    my_settings.visual.login_page = default_page
 
 
-def set_to_default(page, my_settings):
-    view = navigate_to(my_settings, 'Visual')
-    view.tabs.visual.start_page.show_at_login.fill(page)
-    view.tabs.visual.save.click()
+@pytest.fixture(scope="module")
+def set_landing_page(appliance, my_settings, start_page):
+    my_settings.visual.login_page = start_page
+    appliance.server.logout()
 
 
-@test_requirements.settings
-@pytest.mark.parametrize(
-    "start_page",
-    LANDING_PAGES if current_version() > "5.10" else LANDING_PAGES_5_9,
-    scope="module",
+@pytest.mark.parametrize("start_page", ALL_LANDING_PAGES, scope="module")
+@pytest.mark.uncollectif(
+    lambda start_page, appliance: (
+        (appliance.version < "5.11" and start_page in PAGES_NOT_IN_510)
+        or (appliance.version > "5.11" and start_page in PAGES_NOT_IN_511)
+    )
 )
-def test_landing_page_admin(start_page, appliance, my_settings, request):
+@test_requirements.settings
+def test_landing_page_admin(
+    appliance, request, set_default_page, set_landing_page, start_page
+):
     """
-            This test checks the functioning of the landing page; 'Start at Login'
-            option on 'Visual' tab of setting page for administrator. This test case doesn't
-            check the exact page but verifies that all the landing page options works properly.
+    This test checks the functioning of the landing page; 'Start at Login'
+    option on 'Visual' tab of setting page for administrator. This test case
+    check the exact page and verifies that all the landing page options works properly.
 
     Polarion:
         assignee: pvala
         casecomponent: Configuration
-        caseimportance: medium
         initialEstimate: 1/8h
         tags: settings
+        setup:
+            1. Navigate to `My Settings` > `Visual` > `Start Page` and fill `Show at login`.
+            2. Logout and Login
+        testSteps:
+            1. Check the page displayed.
+        expectedResults:
+            1. The page displayed must be same as what  was set.
+
+    Bugzilla:
+        1656722
     """
-    request.addfinalizer(lambda: set_to_default('Cloud Intel / Dashboard', my_settings))
-    assert set_landing_page(start_page, appliance, my_settings)
+    logged_in_page = appliance.server.login_admin()
+    steps = map(lambda x: x.strip(), start_page.split("/"))
```

Comment:
Please use comprehension here instead of map+lambda.
`steps = [x.strip() for x in start_page.split('/')]`

Reply:
You also can avoid the lambda by passing the strip function to `map`

`steps = map(str.strip, start_page.split('/'))`

Reply:
@valaparthvi this can be in a followup PR, what you have here is good otherwise and I'd like to merge. :)